test/linalg/test_linalg.fypp

```diff
@@ -552,6 +562,336 @@ contains
             "trace(h) == sum(c(0:nd:2)) failed.")
 
     end subroutine test_trace_int64
+    
+    subroutine test_kronecker_product_rsp(error) 
+      !> Error handling
+      type(error_type), allocatable, intent(out) :: error
+      integer, parameter :: m1=1, n1=2, m2=2, n2=3
+      real(sp), parameter :: tol = 1.e-6
+
+      real(sp) :: A(m1,n1), B(m2,n2)
+      real(sp) :: C(m1*m2,n1*n2), expected(m1*m2,n1*n2), diff(m1*m2,n1*n2)
+
+      integer :: i,j
+
+      do j=1, n1
+         do i=1, m1
+            A(i,j) = i*j ! A = [1, 2]
+         end do
+      end do
+
+      do j=1, n2
+         do i=1, m2
+            B(i,j) = i*j ! B = [[1, 2, 3], [2, 4, 6]]
+         end do
+      end do
+
+      C = kronecker_product(A,B)
+      expected = transpose(reshape([1,2,3, 2,4,6, 2,4,6, 4, 8, 12], [m2*n2, m1*n1]))
```

Comment:
This matrix `expected` could be a `real, parameter`

Reply:
I could make it a parameter. Should I break with the outer_product convention to do so?

Reply:
I don't think there is a convention for such things.

Reply:
In my push I made `expected` a parameter.